sw/device/silicon_creator/lib/cert/dice_cwt.c

```diff
@@ -81,7 +208,93 @@ rom_error_t dice_cdi_1_cert_build(hmac_digest_t *owner_measurement,
                                   cert_key_id_pair_t *key_ids,
                                   ecdsa_p256_public_key_t *cdi_1_pubkey,
                                   uint8_t *cert, size_t *cert_size) {
-  // TODO(lowRISC/opentitan:#24281): implement body
+  // Build Subject public key structure
+  size_t cose_key_size = sizeof(cose_key_buffer);
+  cwt_cose_key_values_t cwt_cose_key_params = {
+      .pub_key_ec_x = (uint8_t *)cdi_1_pubkey->x,
+      .pub_key_ec_x_size = sizeof(cdi_1_pubkey->x),
+      .pub_key_ec_y = (uint8_t *)cdi_1_pubkey->y,
+      .pub_key_ec_y_size = sizeof(cdi_1_pubkey->y),
+  };
+  HARDENED_RETURN_IF_ERROR(cwt_cose_key_build(
+      &cwt_cose_key_params, &cose_key_buffer[0], &cose_key_size));
+
+  // Try to generate DiceChainEntryPayload
+  fill_dice_id_string((uint8_t *)(&key_ids->endorsement->digest[0]),
+                      &issuer[0]);
+  fill_dice_id_string((uint8_t *)(&key_ids->cert->digest[0]), &subject[0]);
```

Comment:
```suggestion
  fill_dice_id_string((uint8_t *)(key_ids->endorsement->digest), issuer);
  fill_dice_id_string((uint8_t *)(key_ids->cert->digest), subject);
```
same here, and throughout actually.